src/Payments.sol

```diff
@@ -832,8 +843,17 @@ contract Payments is
 
         // Credit operator (if commission exists)
         if (operatorCommission > 0) {
-            Account storage operatorAccount = accounts[token][operator];
-            operatorAccount.funds += operatorCommission;
+            // Ensure the contract has enough funds to pay the operator commission
+            require(
+              IERC20(token).balanceOf(address(this)) >= operatorCommission,
```

Comment:
@SarveshLimaye I don't think understand why we do this now. We want to keep the funds for the commission in the payments contract account of this `serviceFeeRecipient `. Why are we doing an ERC-20 transfer ? Can you explain this decision in detail ?

Reply:
@SarveshLimaye Yes, let's keep the funds in the Payments contract for now. Owners of the account can always call `withdraw` if they wanna move funds out of the contract.